src/peer/identity.js

```diff
@@ -28,25 +27,24 @@ function loadIdentity (filePath: string): Promise<PeerId> {
 }
 
 
-function loadOrGenerateIdentity(filePath: string): PeerId {
-  let peerId
-  try {
-    loadIdentity(filePath)
-  } catch(err) {
-    console.log(`Could not load from ${filePath}, generating new PeerId...`)
-    peerId = generateIdentity()
-    saveIdentity(peerId, filePath)
-  }
-
-  return peerId
+function loadOrGenerateIdentity(filePath: string): Promise<PeerId> {
+    return loadIdentity(filePath)
+      .catch(err => {
+        console.log(`Could not load from ${filePath}, generating new PeerId...`)
+        return generateIdentity()
+      })
+      .then(id => {
+        saveIdentity(id, filePath)
+        return id
+      })
 }
 
-function inflateMultiaddr(multiaddrString: string): PeerInfo {
+function inflateMultiaddr(multiaddrString: string): Promise<PeerInfo> {
     const multiaddr = Multiaddr(multiaddrString)
-    const peerInfo = new PeerInfo()
-    peerInfo.multiaddr.add(multiaddr)
-
-    return peerInfo
+    return PeerInfo.create().then(peerInfo => {
```

Comment:
so the issue is that I'm actually not 100% sure this is correct, see here: https://github.com/libp2p/js-peer-info/commit/5dc116221aa603b857b395336a3e8bbc2b80a29c#commitcomment-19831559

notice that the bytes in the peerid don't match the passed in multiaddr? no idea if this an unimportant omission, weird semantics, or a security hole where anyone (?) can pretend to be QmTVsocFCSEdPyM8dZ734GRhjvvmYyL9fShyezVkbPj17E and you won't know

I ended up doing this instead:

```js
    const multiaddr = Multiaddr(multiaddrString)
    const ipfsIdB58String =multiaddr.stringTuples().filter((tuple) => {
      if (tuple[0] === IPFS_CODE) {
        return true
      }
    })[0][1]
    const peerId = PeerId.createFromB58String(pubKey)
    const peerInfo = new PeerInfo(peerId)
    peerInfo.multiaddr.add(multiaddr)
```